tsl/src/compression/compression.c

```diff
@@ -2483,6 +2488,24 @@ decompress_chunk_walker(PlanState *ps, List *relids)
 							 errhint("Set timescaledb.enable_dml_decompression to TRUE.")));
 
 				decompress_batches_for_update_delete(current_chunk, predicates);
+
+				/* This is a workaround specifically for bitmap heap scans:
+				 * during node initialization, initialize the scan state with the active snapshot
+				 * but since we are inserting data to be modified during the same query, they end up
+				 * missing that data by using a snapshot which doesn't account for this decompressed
+				 * data. To circumvent this issue, we change the internal scan state to use the
+				 * transaction snapshot and execute a rescan so the scan state is set correctly and
+				 * includes the new data.
+				 */
+				if (should_rescan)
+				{
+					ScanState *ss = ((ScanState *) ps);
+					if (ss && ss->ss_currentScanDesc)
+					{
+						ss->ss_currentScanDesc->rs_snapshot = GetTransactionSnapshot();
```

Comment:
Here we are changing state of scan node. I think its safe to reset to its original state once we finish rescanning.

Reply:
I don't think its necessary, we do this will all other scan type by changing the estate snapshot. They are not reverted back once we revert the estate snapshot.